src/content/docs/en/guides/middleware.mdx

```diff
@@ -11,10 +11,15 @@ This also allows you to set and share request-specific information across endpoi
 
 Middleware is available in both SSG and SSR Astro projects.
 
-:::note
-On all prerendered pages, the middleware runs only during the **build process** to create static pages and does not run in the deployed website. This applies to all pages in a `static` (SSG) project. This also includes prerendered pages in `hybrid` mode (default) and any pages in `server` mode with `export const prerender = true`.
+### Timing of middleware execution
+By default, astro builds your project into a static site. In this [mode](https://docs.astro.build/en/basics/rendering-modes/), the middleware runs during the build as well. This allows the pages access to [`locals` object](#storing-data-in-contextlocals). Additionally, the static files can be served immediately since no server-side code is run on-request, ensuring fast response times.
```

Comment:
Users ask "why not run middleware only on-request for static pages". So I tried to give the 2 reasons here:
- you'll need `Astro.locals`, and
- you probably don't want server-side code to run on-request if it can run ahead of time for performance.

Reply:
OK, so I see this less about timing and more about "why would I run middleware on static pages?"

In that framing, I can see adding something here. Let me try something!